sklearn/ensemble/tests/test_gradient_boosting.py

```diff
@@ -870,11 +872,11 @@ def test_warm_start_oob_switch(Cls):
 
     assert_array_equal(est.oob_improvement_[:100], np.zeros(100))
     assert_array_equal(est.oob_scores_[:100], np.zeros(100))
-    assert est.oob_scores_[-1] == pytest.approx(est.oob_score_)
 
     # the last 10 are not zeros
-    assert_array_equal(est.oob_improvement_[-10:] == 0.0, np.zeros(10, dtype=bool))
-    assert_array_equal(est.oob_scores_[-10:] == 0.0, np.zeros(10, dtype=bool))
+    assert est.oob_improvement_[-10:].all() != 0.0
+    assert est.oob_scores_[-10:].all() != 0.0
```

Comment:
I don't think this is correct or at least it is not easier to understand. `est.oob_improvement[-10:].all()` should output some bool and you compare it with `0.0`.

I assume that we could have `assert est.oob_improvement[-10:].all().all()`. Since the scores are always positive, I think that the version of @betatim is easier to read and understand.

Reply:
Okay, I will change it to @betatim's version.

Reply:
It should be `assert (est.oob_scores_[-10:] != 0.0).all()`. This first produces an array of booleans indicating where the values of `est.oob_scores_[-10:]` are not equal to zero, then we call `.all()` on that array to find out if all elements are `True`.

Reply:
Yes. @betatim I have changed it to that.